python/ray/train/v2/api/data_parallel_trainer.py

```diff
@@ -210,19 +210,29 @@ def _initialize_and_run_controller(self, **controller_init_kwargs) -> Result:
 
     def _register_sigint_handler(self, controller: TrainController):
         """Register SIGINT handler so user Ctrl C gracefully aborts run."""
+        sigint_count = 0
 
         def sigint_handler(signum, frame):
-            try:
+            logger.info(
+                "Received SIGINT. Gracefully aborting the training run — this "
+                "may take a few seconds. To forcefully abort immediately, you "
+                "can send a different signal, such as SIGKILL."
+            )
+            nonlocal sigint_count
+            sigint_count += 1
+            if sigint_count >= 3:
                 logger.info(
-                    "Received SIGINT. Gracefully aborting the training run — this "
-                    "may take a few seconds. To forcefully abort immediately, you "
-                    "can send a different signal, such as SIGKILL."
+                    "Received SIGINT at least 3 times. "
+                    "Forcefully aborting the training run."
                 )
-                ray.get(controller.abort.remote())
-            except ray.exceptions.ActorDiedError:
-                # We catch the error and exit 0 to indicate graceful termination.
-                # However, for some reason the process still exits with 1.
                 sys.exit(0)
+            if sigint_count <= 1:
```

Comment:
nit: Do we want to try going through the abort flow when `sigint_count = 2`?

Reply:
I don't have any strong opinions but that's what it was before https://github.com/ray-project/ray/pull/54188#discussion_r2180667253 CC @justinvyu